### PR TITLE
Add deterministicy check for array type CBOR bytes

### PR DESCRIPTION
**Pull request by sonkkeli**

### CODE REVIEW

File: go/internal/cbor/deterministic.go [hunk 1]
```diff
@@ -123,3 +124,24 @@ func textOrByteStringDeterministic(input []byte) (int, error) {
 
 	return uintLen + int(stringLen), nil
 }
+
+// arrayDeterministic returns length of the CBOR array in bytes and checks that each item on it follows the deterministic principles.
+func arrayDeterministic(input []byte) (int, error) {
+	lenOfNumOfItems, numOfItems, err := unsignedIntegerDeterministic(input[0:])
```
Comment (irori) [inline go/internal/cbor/deterministic.go:130]:
```suggestion
	lenOfNumOfItems, numOfItems, err := unsignedIntegerDeterministic(input)
```

File: go/internal/cbor/deterministic_test.go [hunk 2]
```diff
@@ -188,6 +188,48 @@ func TestByteStringIsNotDeterministic(t *testing.T) {
...
+	cborSequenceArray := multiappend(longArr, longArr)
+
+	for _, testCase := range [][]byte{longArr, cborSequenceArray} {
+		err := Deterministic(testCase)
+		if err != nil {
+			t.Error("Deterministically encoded CBOR array should not return false for deterministicy.")
+		}
+	}
+}
+
```
Comment (irori) [inline go/internal/cbor/deterministic_test.go:232]: Can we have some negative test cases? (e.g. an array containing a non-deterministically-encoded CBOR item).

Comment (sonkkeli) [reply go/internal/cbor/deterministic_test.go:227]: I added one with non-deterministic item and another with too little items :)

**Comment by sonkkeli** [comment]
PTAL

**Review by irori — APPROVED** [review]

**Pull request merged**